### PR TITLE
Angular 13 migration

### PR DESCRIPTION
**Pull request by fgravin**
From 12 to **13**
Additional required fixes

Command to run the update
```
npm run update
```
Note that

- [x] I passed the `target` TS build to `es2020` even though it could not be definitive, I wanted a simple workaround to the `BigInt` issue, maybe we could need a polyfill IDK but maybe this `target` is ok.
- [x] I had to update the tailwind config and use `jit` `mode` to purge the CSS on the fly even for DEV mode, cause there is no more `ENV` that tells if we are in production or not (was `process.env.TAILWIND_MODE` before).
- [x] Storybook stories are now compiled with TS, so I had to update the `tsconfig.ts` file and fix stories issues.

### CODE REVIEW

**Review by jahow — APPROVED** [review]
Very nice, thanks for the migration and added tests!

Could you maybe copy/paste the commands that you use, for future reference?

**Comment by fgravin** [comment]
Production build fails

```
Error: Optimization error [main.042500c9f12672f2.js]: Error: Transform failed with 5 errors:
main.042500c9f12672f2.js:97259:17: ERROR: Big integer literals are not available in the configured target environment ("es2015")
main.042500c9f12672f2.js:97260:17: ERROR: Big integer literals are not available in the configured target environment ("es2015")
main.042500c9f12672f2.js:97426:57: ERROR: Big integer literals are not available in the configured target environment ("es2015")
main.042500c9f12672f2.js:98218:18: ERROR: Big integer literals are not available in the configured target environment ("es2015")
main.042500c9f12672f2.js:98218:62: ERROR: Big integer literals are not available in the configured target environment ("es2015")
    at failureErrorWithLog (/home/fgravin/dev/geonetwork-ui/node_modules/@angular-devkit/build-angular/node_modules/esbuild/lib/main.js:1557:15)
    at /home/fgravin/dev/geonetwork-ui/node_modules/@angular-devkit/build-angular/node_modules/esbuild/lib/main.js:1346:29
    at /home/fgravin/dev/geonetwork-ui/node_modules/@angular-devkit/build-angular/node_modules/esbuild/lib/main.js:637:9
    at handleIncomingPacket (/home/fgravin/dev/geonetwork-ui/node_modules/@angular-devkit/build-angular/node_modules/esbuild/lib/main.js:734:9)
    at Socket.readFromStdout (/home/fgravin/dev/geonetwork-ui/node_modules/@angular-devkit/build-angular/node_modules/esbuild/lib/main.js:604:7)
    at Socket.emit (events.js:400:28)
    at addChunk (internal/streams/readable.js:293:12)
    at readableAddChunk (internal/streams/readable.js:267:9)
    at Socket.Readable.push (internal/streams/readable.js:206:10)
    at Pipe.onStreamRead (internal/stream_base_commons.js:188:23)



 —————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Ran target build for project datahub and 1 task(s) it depends on (2m)
 
    ✖    1/2 failed
    ✔    1/2 succeeded [0 read from cache]
 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! geonetwork-ui@0.0.0 build: `ng build "datahub"`
npm ERR! Exit status 1

```

File: tailwind.base.config.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 module.exports = {
+  mode: 'jit',
   purge: {
-    enabled: process.env.TAILWIND_MODE === 'build',
+    enabled: true,
```
Comment (jahow) [inline tailwind.base.config.js:4]: Is that still needed with JIT mode?

File: libs/ui/widgets/src/lib/popup-alert/popup-alert.component.stories.ts [hunk 2]
```diff
@@ -35,19 +35,20 @@ export default {
...
+const content =
+  'Something went wrong during a task that was probably too complicated, you should probably <a href="https://www.google.com">look for answers</a> and come back!'
+
 const Template: Story<PopupAlertComponentWithContent> = (
   args: PopupAlertComponentWithContent
 ) => ({
   component: PopupAlertComponent,
   props: args,
-  template: `<gn-ui-popup-alert [icon]='icon' [position]='position'>${args.content}</gn-ui-popup-alert>`,
+  template: `<gn-ui-popup-alert [icon]='icon' [position]='position'>${content}</gn-ui-popup-alert>`,
```
Comment (jahow) [inline libs/ui/widgets/src/lib/popup-alert/popup-alert.component.stories.ts:46]: Hmm, shame that we're not able to change the content of the popup anymore :/

**Review by jahow — APPROVED** [review]
Two small questions, haven't tested it yet. There is nothing else needed to enable JIT mode in tailwind?

File: libs/ui/widgets/src/lib/popup-alert/popup-alert.component.stories.ts (hunk 2 above)
Comment (fgravin) [reply libs/ui/widgets/src/lib/popup-alert/popup-alert.component.stories.ts:46]: content is not an input so it doesn't compile

File: tailwind.base.config.js (hunk 1 above)
Comment (fgravin) [reply tailwind.base.config.js:4]: No we can get rid off it, it is the default I guess.
If we remove it, i guess we need to update all apps/libs tailwind config which extend this object, not a big deal.

**Pull request merged**